File: 2025_09/leetcode2025-09-08.py
from typing import List

class Solution:
    def getNoZeroIntegers(self, n: int) -> List[int]:
        for i in range(1, n//2+1):
            j = n-i
            if '0' not in str(i) and '0' not in str(j):
                return [i,j]
        return []

if __name__ == "__main__":
    s = Solution()
    n = 2
    print(s.getNoZeroIntegers(n))

File: leetcode2025/2025_09/leetcode2025-09-08.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
private:
    bool is_zero_in(int a){
        while (a){
            if (a%10 == 0){
                return true;
            }
            a /= 10;
        }
        return false;
    }
public:
    vector<int> getNoZeroIntegers(int n) {
        for (int i=1; i<=n/2; i++){
            int j = n-i;
            if (!is_zero_in(i) && !is_zero_in(j)){
                return {i,j};
            }
        }
        return {};
    }
};

int main(){
    Solution s;
    int n = 2;
    vector<int> res = s.getNoZeroIntegers(n);
    cout << res[0] << " " << res[1] << endl;
}
